Adds glossary entries for 'sos report' and 'sosreport' (#327)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/glossary_terms_conventions/general_conventions/s.adoc

*See also*:

[discrete]
[[sos-report]]
==== image:images/yes.png[yes] sos report (noun)
*Description*: An _sos report_ is an archive of system information, including configuration and diagnostic information.
Red{nbsp}Hat Support engineers review sos reports when troubleshooting system problems.
The command to generate an sos report depends on the RHEL version:

* In RHEL 8 and later, run the `sos report` command (that is, the `sos` command with the `report` argument).
* In RHEL 7 and earlier, run the `sosreport` command.

> NOTE: The `sos` command is an acronym for "son of sysreport", and the official pronunciation is "ess-oh-ess". Therefore, use the indefinite article "an".

Reviewer: @bergerhoffer @bredamc @bburt-rh This language is mostly fine and a good effort. I would have added this edit directly to the text, but I think we'll need to discuss it here first:

- Question 1. Should we say that if you refer to "sosreport" or "sos report", it should always be written as code in a monospaced font? That's how I usually see it in the docs. We probably need to make that explicit.
- Question 2. OpenShift talks about generating a "sosreport" here: https://docs.openshift.com/container-platform/4.13/support/gathering-cluster-data.html#about-sosreport_gathering-cluster-data How do you know if "sos report" or "sos report" is correct if you're not working in RHEL? Would OpenShift 4.13 for sure be on RHEL8 or is it variable? How do we handle "sos report" in that situation?

Thanks,
Ingrid

> NOTE: The `sos` command is an acronym for "son of sysreport", and the official pronunciation is "ess-oh-ess". Therefore, use the indefinite article "an".

Author: I updated the PR to split `sos report` and `sosreport` into two separate "Use with caution" entries. 

I also included the word "archive", because I think the main problem area is what to call the file that's created when you run the command, and "archive" indicates that it's an aggregate file.

Let me know what you think!

> NOTE: The `sos` command is an acronym for "son of sysreport", and the official pronunciation is "ess-oh-ess". Therefore, use the indefinite article "an".

Reviewer 2: Per the SSG meeting, I think we wanted to update the entries to remove the word "archive" after, does that sound right?

> NOTE: The `sos` command is an acronym for "son of sysreport", and the official pronunciation is "ess-oh-ess". Therefore, use the indefinite article "an".

Author: Yes, and include a mention of the term that is used for the other RHEL version (in each term entry). It's on my list!

*Use it*: yes

> *Incorrect forms*: sosreport (correct only for the command name in RHEL 7 and earlier), SOSreport, SOS report, or any other variation

Reviewer 3: ```suggestion
*Incorrect forms*: sosreport (incorrect except when referring to the command name in RHEL 7 and earlier), SOSreport, SOS report, or any other variation
```
Maybe better to say that it's always incorrect except for this one usage case? WDYt?

Author: Great point, @bburt-rh -- I've made that change, thanks!

*See also*:

[discrete]
[[sos-report]]

> ==== image:images/caution.png[with caution] sos report (noun, adjective)

Author: Removed the references to "adjective".

*Description*: In RHEL 8 and later, an `_sos report_` is a collection of files that contain configuration details, system information, and diagnostic data.
Red{nbsp}Hat Support engineers review an `sos report` when troubleshooting system problems.

Write as shown: two words, monospace font.

> Optionally, append a noun such as _archive_ or _file_, especially when referring to multiple `_sos report_` files.

Author: Removed the unnecessary italics :)

(In RHEL 7 and earlier, use the one-word term `_sosreport_` instead of `_sos report_`.)

> To generate an `sos report` archive, run the `sos report` command (that is, the `sos` command with the `report` argument).

Reviewer 4: Is this an example? An example heading or some sort of flag should be provided.

Consider swapping out parentheses for a semicolon as the semicolon provides a better contrast with the transitional phrase that follows.

> To generate an `sos report` archive, run the `sos report` command; that is, the `sos` command with the `report` argument.

Reviewer 4: What about generating an `sos report` file? Is this a different entity in comparison to an archive. I think an _example_ heading would clear things here.

Reviewer 4: Woth mentioning that on a Linux system one can run the command `man sos` for more info about the command and its options?

Author: 1. It's not an example (although it does show the usage!). Moved the sentence to the initial paragraph and removed the parentheses as suggested.
2. Removed all references to _archive_ and _file_ to avoid confusion.
3. I don't think it's necessary to add this info. The point of the entries is to indicate how to write the terms.

Author: @bergerhoffer Do you agree?

Reviewer 2: I agree that it's not necessary to add the man command.

The `sos` in the command name is an acronym for "son of sysreport", and the official pronunciation is "ess-oh-ess".
Therefore, if you need to use the indefinite article before " `sos report` ", use _an_ not _a_.

*Use it*: yes

*Incorrect forms*: sosreport, SOSreport, SOSREPORT, SOS report, SOS Report, SOS REPORT

*See also*: xref:sosreport[sosreport]

[discrete]
[[sosreport]]

> ==== image:images/caution.png[with caution] sosreport (noun, adjective)

Reviewer 2: The use it is marked as "yes", but this uses the "with caution" icon. Can you update the icon for both of these entries?

Author: Changed the "Use it" text to "with caution" to ensure that people use the right term for their RHEL version.

*Description*: In RHEL 7 and earlier, an `_sosreport_` is a collection of files that contain configuration details, system information, and diagnostic data.
Red{nbsp}Hat Support engineers review an `sosreport` when troubleshooting system problems.

Write as shown: one word, monospace font.
Optionally, append a noun such as _archive_ or _file_, especially when referring to multiple `_sosreport_` files.
(In RHEL 8 and later, use the two-word term `_sos report_` instead of `_sosreport_`.)

To generate an `sosreport` archive, run the `sosreport` command.

The `sos` in the command name is an acronym for "son of sysreport", and the official pronunciation is "ess-oh-ess".
Therefore, if you need to use the indefinite article before " `sosreport` ", use _an_ not _a_.

*Use it*: yes

> *Incorrect forms*: SOSreport, SOSREPORT, sos report, SOS report, SOS Report, SOS REPORT

Author: I don't think we need to include a reference to the IBM product? It doesn't add anything to the explanation of how to write the terms "sos report" or "sosreport", and might introduce confusion?

> *Incorrect forms*: SOSreport, SOSREPORT, sos report, SOS report, SOS Report, SOS REPORT

Author: @bergerhoffer @mportman12 Do you agree?

> *Incorrect forms*: SOSreport, SOSREPORT, sos report, SOS report, SOS Report, SOS REPORT

Reviewer 2: I think we're okay without it. There are probably lots of other things that uses "SOS". and ours is lowercase, and has "report" after it, so there shouldn't be confusion.

> *Incorrect forms*: SOSreport, SOSREPORT, sos report, SOS report, SOS Report, SOS REPORT

Reviewer 4: Thanks for confirming. Maybe add "SOS" to the `Incorrect forms` field?

> *Incorrect forms*: SOSreport, SOSREPORT, sos report, SOS report, SOS Report, SOS REPORT

Author: _(Apologies, I click the wrong button and resolved by mistake!)_

And apologies for disagreeing with you again @dfitzmau but I think it's unnecessary to add "SOS" to the "Incorrect forms" field -- as far as I'm aware, that incorrect form has not been used to refer to the sos report. To me, your suggestion is analogous to adding "BASIC" to the "Incorrect forms" field for the "Basic HTTP authentication" entry. But I'll defer to the majority decision :)

> *Incorrect forms*: SOSreport, SOSREPORT, sos report, SOS report, SOS Report, SOS REPORT

Reviewer 4: NP. I just spotted some of Red Hat's support articles that use the acronym "SoS". Example: https://access.redhat.com/articles/445443

Sometimes doc Jira requests might include a link to article for supplementary information, so if this article were attached, maybe a writer could fall into the trap of using `SoS` in their content. Please mark the comment as "Resolved" if my comment is still not applicable. Thanks!

> *Incorrect forms*: SOSreport, SOSREPORT, sos report, SOS report, SOS Report, SOS REPORT

Author: Ah, good spot! I'll add "SoS report" and "SoS Report" to the list of Incorrect forms! Thanks for your patience, @dfitzmau :)

> *Incorrect forms*: SOSreport, SOSREPORT, sos report, SOS report, SOS Report, SOS REPORT

Author: Done. I'll resolve this conversation, but feel free to reopen it if you'd like to discuss any of your points further :)
